qiita_pet/handlers/admin_processing_job.py

```diff
@@ -91,3 +92,40 @@ def get(self):
 
         # return the json in compact form to save transmit size
         self.write(dumps(results, separators=(',', ':')))
+
+
+class SampleValidation(AdminProcessingJobBaseClass):
+    @coroutine
+    @execute_as_transaction
+    def get(self):
+        self._check_access()
+
+        self.render("sample_validation.html", input=True)
+
+    @execute_as_transaction
+    def post(self):
+
+        # Get user-inputted qiita id and sample names
+        qid = self.get_argument("qid")
+        snames = self.get_argument("snames").split()
+
+        qsnames = list(Study(qid).sample_template)
+        for i, qsname in enumerate(qsnames):
+            if qsname.startswith(qid):
+                qsnames[i] = qsname.replace(f'{qid}.', "", 1)
+
+        blank = []
+        for name in snames:
+            if name.lower().startswith('blank'):
+                blank.append(name)
+                snames.remove(name)
+
```

Comment:
I would consider replacing lines 117-121 with the following two lines below:
blank = [x for x in snames if x.lower().startswith('blank')]
snames = [x for x in snames if 'blank' not in x.lower()]

These are called list comprehensions and internally they are faster than anything within a 'for name in snames' loop will be. Currently you're iterating through a list and modifying it at the same time, which can lead to unpredictable behavior. The list comprehensions above create a new list based on snames without altering it. Instead the second line will replace the old snames with the new one once the loop has completed.

Reply:
Replaced lines 117-121 with code suggested above